src/spectral.ts

```diff
@@ -240,6 +243,20 @@ export class Spectral {
 
     return resolveOpts;
   };
+
+  private static _generateUnrecognizedFormatError(parsedResult: IParsedResult): IRuleResult {
+    return {
+      range: parsedResult.getLocationForJsonPath(parsedResult.parsed, [], true)?.range || {
+        start: { character: 0, line: 0 },
+        end: { character: 0, line: 0 },
+      },
+      message: 'Provided file format does not match any of the registered formats',
```

Comment:
Reads a little weird - the user is not providing a file format (to me reads like it's some format option I provided) - they're providing a document (file, url, etc) that has a format. Perhaps the following?

`The provided document does not match any of the registered formats [openapi, markdown]` - bonus if can get the registered format names in there to give end user a bit more context.